pkg/storage/utils/decomposedfs/node/node.go

```diff
@@ -963,29 +907,25 @@ func (n *Node) IsDisabled() bool {
 // GetTreeSize reads the treesize from the extended attributes
 func (n *Node) GetTreeSize() (treesize uint64, err error) {
 	var b string
-	if b, err = xattrs.Get(n.InternalPath(), xattrs.TreesizeAttr); err != nil {
+	if b, err = n.Xattr(xattrs.TreesizeAttr); err != nil {
 		return
 	}
 	return strconv.ParseUint(b, 10, 64)
 }
 
 // SetTreeSize writes the treesize to the extended attributes
 func (n *Node) SetTreeSize(ts uint64) (err error) {
-	return n.SetMetadata(xattrs.TreesizeAttr, strconv.FormatUint(ts, 10))
+	return n.SetXattr(xattrs.TreesizeAttr, strconv.FormatUint(ts, 10))
 }
 
 // SetChecksum writes the checksum with the given checksum type to the extended attributes
 func (n *Node) SetChecksum(csType string, h hash.Hash) (err error) {
-	return n.SetMetadata(xattrs.ChecksumPrefix+csType, string(h.Sum(nil)))
+	return n.SetXattr(xattrs.ChecksumPrefix+csType, string(h.Sum(nil)))
 }
 
 // UnsetTempEtag removes the temporary etag attribute
 func (n *Node) UnsetTempEtag() (err error) {
-	err = xattrs.Remove(n.InternalPath(), xattrs.TmpEtagAttr)
-	if xattrs.IsAttrUnset(err) {
-		return nil
-	}
```

Comment:
This attrunset check is also removed. Not sure if this is critical

Reply:
This error is only logged further up the stack, and it shouldn't occur in the first place.